internal/controller/api/job_receiver.go

```diff
@@ -1,23 +1,80 @@
 package api
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/gorilla/mux"
 
 	//	"context"
 	"encoding/json"
-	"io"
-	"io/ioutil"
 	"log"
 	"net/http"
 
 	"github.com/google/uuid"
 
 	"github.com/RedHatInsights/platform-receptor-controller/internal/controller"
+	"github.com/go-playground/validator/v10"
 	"github.com/redhatinsights/platform-go-middlewares/identity"
 
 	kafka "github.com/segmentio/kafka-go"
 )
 
+type badRequest struct {
+	status int
+	msg    string
+}
+
+func (br *badRequest) Error() string {
+	return fmt.Sprintf("%d: %s", br.status, br.msg)
+}
+
+func decodeJSON(w http.ResponseWriter, req *http.Request, job interface{}) error {
```

Comment:
What do you think about only passing in the req.Body (io.Reader?), job struct instance and just return an error?  This way the http request/response handling is only contained in the main handler method and this method just deals with decoding the json.

I think its fine to just return StatusBadRequest for any of the json parsing/handling failures.

Reply:
Sure I can do that. Do you mean take out the UnprocessableEntity errors and just use BadRequest?

Reply:
Yeah, i think its fine to just return BadRequest back to the client in the case of the client sending in bad/invalid json.